experiments/dsl/src/main/scala/org/apache/gearpump/streaming/dsl/plan/OpTranslator.scala

```diff
@@ -182,39 +179,43 @@ object OpTranslator {
     }
   }
 
-  class SourceTask[T, OUT](source: TraverseType[T], operator: Option[SingleInputFunction[T, OUT]], taskContext: TaskContext, userConf: UserConfig) extends Task(taskContext, userConf) {
+  class SourceTask[T, OUT](source: TypedDataSource[T], operator: Option[SingleInputFunction[T, OUT]], taskContext: TaskContext, userConf: UserConfig) extends Task(taskContext, userConf) {
 
     def this(taskContext: TaskContext, userConf: UserConfig) = {
       this(
-        userConf.getValue[TraverseType[T]](GEARPUMP_STREAMING_SOURCE)(taskContext.system).get,
+        userConf.getValue[TypedDataSource[T]](GEARPUMP_STREAMING_SOURCE)(taskContext.system).get,
         userConf.getValue[SingleInputFunction[T, OUT]](GEARPUMP_STREAMING_OPERATOR)(taskContext.system),
         taskContext, userConf)
     }
 
     override def onStart(startTime: StartTime): Unit = {
+      source.open(taskContext, Some(startTime.startTime))
       self ! Message("start", System.currentTimeMillis())
     }
 
     override def onNext(msg: Message): Unit = {
       val time = System.currentTimeMillis()
-      source.foreach(msg => {
+      source.read(1).foreach(msg => {
```

Comment:
There supposed to be a batch size configuration, but I'm not sure we should expose this parameter on the ```source``` interface.